### PR TITLE
Page split logic adjustment

### PR DESCRIPTION
**Pull request by mpolomdeepsense**
- [x] ~~the default for `pdf_split_page` is set to `True`~~ the default for `pdf_split_page` is set to `False`
- [x] introduced contants for controlling split size: `MIN_PAGES_PER_THREAD` and `MAX_PAGES_PER_THREAD`
- [x] number of threads used for processing each PDF split can be configured through `split_pdf_threads` parameter instead of env var
- [x] evenly distributed split sizes
- [x] basic tests for the logic mentioned above

### CODE REVIEW

**Review by awalker4 — APPROVED** [review]
LGTM! Although we had another game time decision to not split by default just yet. We'll set to true when pay per page is fully released.

File: README.md [hunk 1]
```diff
@@ -164,27 +166,30 @@ See the [general partition](docs/sdk/models/shared/partitionparameters.md) page
 
 ### Splitting PDF by pages
 
-In order to speed up processing of long PDF files, set `splitPdfPage` parameter to `true`. It will cause the PDF to be split page-by-page at client side, before sending to API, and combining individual responses as single result. This will work only for PDF files, so don't set it for other types of files.
+In order to speed up processing of long PDF files they are split into smaller batches at client side, before sending to API, and combining individual responses as single result. This will work only for PDF files, so don't set it for other types of files. Size of each batch is determined internally and it can vary between 2 and 20 pages per split.
+
+**Warning:** *this feature causes the `parent_id` metadata generation in elements to be disabled, as it requires having context of multiple pages.*
```
Comment (awalker4) [inline README.md:171]: This can be removed. Parent id is still being returned, and it should be equivalent now with the use of `starting_page_number`

Comment (mpolomdeepsense) [reply README.md]: Changed

File: docs/sdk/models/shared/partitionparameters.md [hunk 2]
```diff
@@ -26,6 +26,8 @@
...
...                                                                                                                                                                                                                                                                          | :heavy_minus_sign:                                                                                                                                                                                                                                                                                                   | Number of threads used when sending requests of splitted PDF. Ignored on backend.                                                                                                                                                                                                                                    |                                                                                                                                                                                                                                                                                                                      |
```
[the remark this answers is not in the document]
Comment (mpolomdeepsense) [reply docs/sdk/models/shared/partitionparameters.md]: Renamed

**Review by awalker4 — APPROVED** [review]
Looks good!

**Comment by mpolomdeepsense** [comment]
@awalker4 

Not sure if I should merge it right now, the api hasn't been updated yet with the new version of the openapi spec that Hubert fixed. That means the generated client won't be working properly.

There is also a matter of publishing the new version of the client, I believe it's disabled so it won't be published automatically.

**Comment by awalker4** [comment]
Yeah, since we aren't publishing yet id say go for it and then we'll get the openapi fix deployed soon

**Pull request merged**